### DISCUSSION TITLE
UI/UX ideas to consider

### DISCUSSION THREAD

**Discussion by cafloyd** [the-collab-lab/tcl-22-smart-shopping-list#29] [Ideas]
Space for brainstorming for our last 2 weeks/Issues #13 and #14

**Comment by cafloyd** [comment]
We'll need to make sure to display the user's token somewhere so they can share it. It could be nice to use something like [React-copy-to-clipboard](https://www.npmjs.com/package/react-copy-to-clipboard) for ease of use. (See PR #25 )

**Comment by dariatsvetkova** [comment]
In Notistack, the notifications have icons in them. The icon on the error message looks a lot like a "close" button. I tried clicking on it a bunch of times before I realized it was not a button😄. Maybe we should consider a custom icon for this one.

![image](https://user-images.githubusercontent.com/68360696/115812533-8af00980-a3bf-11eb-91b4-6d1ec0d53610.png)

**Reply by cafloyd** [reply]
Agreed, we didn't realize it wasn't a `close` button until we saw the success message!

**Reply by ablizben** [reply]
I was wondering if there's a time limit we can edit somewhere for how long it shows? Depending on what our UI will look like after we make it all pretty, we might want to change it because when it's on a smaller screen, it blocks our GroceryList link for a while (like 3 seconds hahahaha but we all know how the general user has no patience in today's world).

**Reply by cafloyd** [reply]
I wonder if there are any accessibility implications for this. I think adding a shorter duration for the message is a great idea. It could also be great if it would automatically dismiss the notification if the user clicks/taps anywhere outside of it.

**Comment by cafloyd** [comment]
# Color palette

Potential tools:
- [paintr](http://paintr.io/) (by our very own Daria!)
- [coolors](https://coolors.co/)

**Reply by cafloyd** [reply]
I did a google image search for "shopping list app", and noticed a lot of green in the search results. This kind of made sense to me, since I think people probably associate both produce and money with the color green 
![image](https://user-images.githubusercontent.com/58532369/117556815-f3cfb680-b03a-11eb-9f93-7b25be60a5b7.png)

**Reply by jonathanyeong** [reply]
I also like [Colorhunt](https://colorhunt.co/)!

**Reply by cafloyd** [reply]
Missed this link from #14 : [accessible colors](https://accessible-colors.com/) for checking that colors are accessible

**Reply by cafloyd** [reply]
via @dariatsvetkova , the breakdown of colors we're aiming for: 

30% is main color (green)
10% is accent color (yellow)
rest is neutral colors

**Comment by cafloyd** [comment]
# Font

We should consider [Google Fonts](https://fonts.google.com/)

**Reply by cafloyd** [reply]
**Headings** - Maven Pro
**Body** - Lato

**Comment by cafloyd** [comment]
# Icons

[Canva Icon Library](https://www.canva.com/learn/free-icons-download/)

**Reply by cafloyd** [reply]
We have some Material UI dependencies installed, so that could be a good path to pursue!

**Reply by cafloyd** [reply]
Not quite icons, but [Lottie](https://lottiefiles.com/) has lots of cute open source animations if we want to include any somewhere like our welcome screen!

**Reply by cafloyd** [reply]
Another thought - I can't find the comment now, but I think someone (maybe @dariatsvetkova ?) made a comment that a lot of shopping lists that use images of food look a little awkward. One possible compromise between no images and awkward images would be to give the user an option to choose an icon to associate with the item, and we can provide icons to correspond to typical grocery categories (produce, bread, milk/eggs, etc). Not married to this idea, but just a thought I had!

**Reply by dariatsvetkova** [reply]
@cafloyd definitely, category icons would be a great alternative to photos!

**Reply by ablizben** [reply]
I like that idea! I think it looks cleaner than random stock photos.

**Reply by ablizben** [reply]
Speaking of cute animations, If we have time, depending on how quick the load time is, we could have a little animation (maybe an avocado) that would appear when things are loading - or just when the welcome page is loading. Just an idea.

**Reply by dariatsvetkova** [reply]
@ablizben I love that! We could have a loading screen with an animated avocado.

**Reply by cafloyd** [reply]
[**React Icons**](http://react-icons.github.io/react-icons/)

**Reply by cafloyd** [reply]
SVG Background generators:

- **[BG Jar](https://bgjar.com/icon-grid)**

- **[SVG Backgrounds](https://www.svgbackgrounds.com/)**

**Reply by kkatoch89** [reply]
@cafloyd Yours came out much better than mine lol. They don't have a large selection of icons eh.

**Reply by cafloyd** [reply]
@kkatoch89 😂  they have one for "dermatology/acne" but they don't have like, an apple? or any food??

**Reply by cafloyd** [reply]
I think I actually kind of like this ["liquid cheese"](https://www.svgbackgrounds.com/#liquid-cheese) background? Or maybe I'm just hungry? 😅 🧀

**Comment by cafloyd** [comment]
# Tone

What is the tone you’re going for? What feeling are you trying to convey? Come up with 3 to 5 words that best capture how you want users to perceive your app. Examples:
- Fun
- Clean
- Ironic
- Businesslike
[Many more example words](http://static1.squarespace.com/static/5723ff8086db4345e722e7be/573248885f21e0a9e45338ef/57b6a206579fb39bc0e08e2a/1471586826621//img.jpg)

**Reply by cafloyd** [reply]
This is such a great idea, @dariatsvetkova ! I'm willing to go along with whatever the group wants, but some words that appealed to me were
- breezy
- sleek
- inviting

**Comment by cafloyd** [comment]
# Mood board

Create a mood board. A mood board helps everyone get on the same page in terms of the direction in which you want to take your design. Add elements to your mood board that represents the tone you’re trying to achieve.

Tools/Resources:
- [Milanote mood board guide](https://milanote.com/guide/moodboard)
- [Canva mood board guide](https://www.canva.com/learn/make-a-mood-board/)

**Reply by cafloyd** [reply]
Linking our [Figma mood board](https://www.figma.com/file/VH8Z8iouh7F9zBBv1kbuq5/Shopping-App-design?node-id=0%3A1) here for easy access

**Comment by dariatsvetkova** [comment]
@ablizben @cafloyd @kkatoch89 as discussed, let's start with a mood board.

Here is a shared document in Figma where we can collaborate: https://www.figma.com/file/VH8Z8iouh7F9zBBv1kbuq5/Shopping-App-design?node-id=0%3A1

Pick a corner and put things there:
* Your name
* 3-5 words for the tone you'd like to use - choose from [this list](https://images.squarespace-cdn.com/content/5723ff8086db4345e722e7be/1471586822426-OBVFFH7NM3W2YUHA8R5S/?content-type=image%2Fjpeg)
* Inspirational photos and images that capture the mood you like - choose from anywhere, for example [Dribbble](https://dribbble.com/), [Behance](https://www.behance.net/), [Unsplash](https://unsplash.com/)
* Screenshots of apps and websites that you like - for example, from https://www.siteinspire.com/ or
https://www.hoverstat.es/
* Screenshots of UI elements that you think could be useful - https://collectui.com/designs or https://littlebigdetails.com/

More on working with mood boards: 

> # Mood board
> Create a mood board. A mood board helps everyone get on the same page in terms of the direction in which you want to take your design. Add elements to your mood board that represents the tone you’re trying to achieve.
> 
> Tools/Resources:
> 
> * [Milanote mood board guide](https://milanote.com/guide/moodboard)
> * [Canva mood board guide](https://www.canva.com/learn/make-a-mood-board/)

**Comment by ablizben** [comment]
These are all really great websites that you're pulling from and suggesting, @dariatsvetkova! Thanks so much for that.

**Comment by laurenmbeatty** [comment]
I was just thinking of this. react-helmet is a neat library for managing the content of your document's `head`, for example the title tag:
https://github.com/nfl/react-helmet

helps update this kind of info: 
![image](https://user-images.githubusercontent.com/10378331/118893609-4d9c7000-b8c8-11eb-8144-e459760aa99b.png)

**Comment by dariatsvetkova** [comment]
# LAST WEEK


| Status |  Things to do   | Responsible | Phase |
|---| --- | -------------------------- | ------ |
| :heavy_check_mark: | Global styles aka theme | @kkatoch89 | 1 |
| :heavy_check_mark: | SVG background | @kkatoch89 | 1 |
| :heavy_check_mark: | Desktop design | @dariatsvetkova | 1 |
| :heavy_check_mark: | New component for sharing a token + a github issue with a user story for it | @ablizben | 2 |
| | Github issues | @dariatsvetkova | 2 |
| | *Actually applying styles from Figma:* 
| | App.js | @kkatoch89  | 3 |
| | Welcome + ShareToken | @cafloyd | 3 |
| | ItemList + Item | @ablizben | 3 |
| | Modal for deleting an item | @ablizben  | 3 |
| :heavy_check_mark: | AddItem | @dariatsvetkova | 3 | 
| | New component for sharing the token | @ablizben | 3 |
| | Accessibility (has to be done the last) | @cafloyd | 4 |
| | Animate the logo | @dariatsvetkova  + @kkatoch89 | 4 |

If you get stuck or overwhelmed with the amount of work, let everyone know and we'll figure it out!

**Reply by ablizben** [reply]
@dariatsvetkova @cafloyd @kkatoch89 Just realized that we don't have a reusable component for the top and bottom of all the views besides the welcome page (the three buttons at the bottom and then the top with the logo). Or do we and I'm just sleep-deprived? Hahaha. I can do them if not.

**Reply by dariatsvetkova** [reply]
I was assuming that navigation/background/the white container are a part of the App component since `<nav>` and all other components sit within it.

**Reply by ablizben** [reply]
I literally just forgot all how you made it so on the welcome screen with no token, you can't route to the other pages. *face palm* ignore me!

**Reply by dariatsvetkova** [reply]
No worries!
